src/rgw/rgw_op.h

```diff
@@ -151,7 +151,8 @@ class RGWOp : public DoutPrefixProvider {
   virtual ~RGWOp() = default;
 
   int get_ret() const { return op_ret; }
-
+  virtual off_t get_ofs() {return std::numeric_limits<off_t>::max();}
+  virtual off_t get_end() {return std::numeric_limits<off_t>::max();}
```

Comment:
we shouldn't expose these in RGWOp, especially with defaults like this. lets just extend the existing prefetch interface to include them
```diff
-  virtual bool prefetch_data() { return false; }
+  virtual std::optional<prefetch_range> prefetch_data() { return std::nullopt; }
```

Reply:
I used pointer so it's a little bit different (done)